browser/main/NoteList/index.js

```diff
@@ -390,25 +398,51 @@ class NoteList extends React.Component {
     return pinnedNotes.concat(unpinnedNotes)
   }
 
+  getNoteIndexByKey (noteKey) {
+    return this.notes.findIndex((note) => {
+      if (!note) return -1
+
+      return note.key === noteKey
+    })
+  }
+
   handleNoteClick (e, uniqueKey) {
     const { router } = this.context
     const { location } = this.props
     let { selectedNoteKeys } = this.state
-    const { shiftKeyDown } = this.state
+    const { ctrlKeyDown, shiftKeyDown } = this.state
+    let firstShiftSelectedNoteIndex = -1
 
-    if (shiftKeyDown && selectedNoteKeys.includes(uniqueKey)) {
+    if (ctrlKeyDown && selectedNoteKeys.includes(uniqueKey)) {
       const newSelectedNoteKeys = selectedNoteKeys.filter((noteKey) => noteKey !== uniqueKey)
       this.setState({
         selectedNoteKeys: newSelectedNoteKeys
       })
       return
     }
-    if (!shiftKeyDown) {
+    if (!ctrlKeyDown && !shiftKeyDown) {
       selectedNoteKeys = []
     }
     selectedNoteKeys.push(uniqueKey)
+
+    if (shiftKeyDown && selectedNoteKeys.length > 0) {
+      const firstSelectedNoteIndex =
+        Math.max(this.getNoteIndexByKey(selectedNoteKeys[0]),
+        this.state.firstShiftSelectedNoteIndex)
+      const lastSelectedNoteIndex = this.getNoteIndexByKey(uniqueKey)
+      const startIndex = Math.min(firstSelectedNoteIndex, lastSelectedNoteIndex)
+      const endIndex = Math.max(firstSelectedNoteIndex, lastSelectedNoteIndex)
```

Comment:
Github suggestion is so cool :smile: 
```suggestion
      const endIndex = firstSelectedNoteIndex > lastSelectedNoteIndex ? firstSelectedNoteIndex : lastSelectedNoteIndex
```

Reply:
Updated